### PR TITLE
Bmm legend make configurable

### PR DESCRIPTION
**Pull request by bradleymeintjes**
Please review

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -1,33 +1,16 @@
...
-	s.Sources = append(s.Sources, logscraper.NewLogSource("sap_notifications", "c:/imqsvar/logs/services/notifications.log", logscraper.JavaLogParser))
-	s.Sources = append(s.Sources, logscraper.NewLogSource("data_model_queries", "c:/imqsvar/logs/services/data-model-queries.log", logscraper.JavaLogParser))
-	s.Sources = append(s.Sources, logscraper.NewLogSource("pcs", "c:/imqsvar/logs/services/pcs/imqs-pcs-ws.log", logscraper.JavaLogParser))
-	s.Sources = append(s.Sources, logscraper.NewLogSource("yellowfin", "c:/imqsvar/yellowfin/appserver/logs/yellowfin.log", logscraper.YellowfinLogParser))
-	s.Sources = append(s.Sources, logscraper.NewLogSource("www_server", "c:/imqsvar/logs/www-server.log", logscraper.GoLogParser))
-	s.Sources = append(s.Sources, logscraper.NewLogSource("www_js", "c:/imqsvar/logs/www-js.log", logscraper.GoLogParser))
-	s.Sources = append(s.Sources, logscraper.NewLogSource("timeseries", "c:/imqsvar/logs/imqs-timeseries.log", logscraper.GoLogParser))
-	s.Sources = append(s.Sources, logscraper.NewLogSource("insite", "c:/imqsvar/logs/insite.log", logscraper.GoLogParser))
+	s.LoadConfiguration("C:/imqsbin/static-conf/logscraper-config.json")
```
Comment (frankvanrietimqs) [inline cmd/main.go]: Not sure what the other go services do, but I'm not too thrilled about the hard-coded config.  Should be read from the cmd line in my opinion.

Comment (bradleymeintjes) [inline cmd/main.go]: agreed it looks like they are command line args in the other go services

File: scraper.go [hunk 2]
```diff
@@ -169,6 +170,28 @@ func NewScraper(hostname, statefile, metalogfile string) *Scraper {
...
+	if errors != nil && len(errors) > 0 {
+		for _, err := range errors {
+			s.logMetaf("Error parsing configuraton file: %v", err)
+		}
+		return
+	}
+
+	s.Sources = append(s.Sources, logSources...)
+	for _, src := range s.Sources {
+		fmt.Printf("Source loaded: %v\n", src)
```
Comment (frankvanrietimqs) [inline scraper.go:190]: Sources_s_ loaded ?

File: config.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	var cfg LogScraperConfig
+	bytes, err := ioutil.ReadFile(file)
+	if err == nil {
+		err = json.Unmarshal(bytes, &cfg)
+	}
+	return &cfg, err
+}
+
+func (config *LogScraperConfig) LogSources() ([]*LogSource, []error) {
+	if config.Sources == nil || len(config.Sources) == 0 {
```
Comment (frankvanrietimqs) [inline config.go]: This doesn't really make sense - if you don't have config something is wrong and we should have aborted execution already or cause an error to  propagated to the caller of this function.

File: scraper.go (hunk 2 above)
Comment (bradleymeintjes) [inline scraper.go:190]: I dont quite understand, this is just printing out each source for debug purposes

Comment (frankvanrietimqs) [inline scraper.go:190]: My bad

File: config.go [hunk 4]
```diff
@@ -32,19 +32,32 @@ func LoadLogScraperConfig(file string) (*LogScraperConfig, error) {
...
+		return nil, err
+	}
+
 	return &cfg, err
 }
 
-func (config *LogScraperConfig) LogSources() ([]*LogSource, []error) {
-	if config.Sources == nil || len(config.Sources) == 0 {
-		return make([]*LogSource, 0), nil
+func (cfg *LogScraperConfig) validate() error {
```
Comment (frankvanrietimqs) [inline config.go]: I really don't think this function is useful. The complexity of the function is equivalent to the invocation of it. Just replace the call to validate with this body?

**Pull request merged**